frontend/app/blueprints/groups/templates/shared.html

```diff
@@ -113,4 +113,169 @@
+                            type="radio"
+                            class="btn-check btn-sm"
+                            name="qc-validation"
+                            id="failed-qc-btn"
+                            autocomplete="off"
+                            value="failed"
+                        >
+                        <label class="btn btn-outline-danger" for="failed-qc-btn"><i class="bi bi-hand-thumbs-down"></i></label>
+                    </div>
+                    <div class="col-auto d-flex flex-row">
+                        <select
+                            id="failed-qc-action"
+                            class="form-select form-select-sm"
+                            name="qc-action"
+                            aria-label="Action on failed sample"
+                            hidden
+                        >
+                            <option selected value="0" disabled>Select action</option>
+                            {% for action in bad_qc_actions %}
+                            <option value="{{action}}">{{action | capitalize}}</option>
+                            {% endfor %}
+                        </select>
+                    </div>
+                    <div id="failed-qc-comment-container" class="row" hidden>
+                        <div class="col-auto">
+                            <label class="text-muted" for="failed-qc-comment">Comment</label>
+                            <textarea
+                                type="text"
+                                class="form-control"
+                                name="qc-comment"
+                                id="failed-qc-comment"
+                            ></textarea>
+                        </div>
+                    </div>
+                </div>
+                <div class="row">
+                    <div class="col-auto">
+                        <button
+                            id="qc-submit-btn"
+                            type="submit"
+                            class="btn btn-sm btn-outline-secondary mt-2"
+                            disabled
+                        >Confirm</button>
+                    </div>
+                </div>
+            </div>
+        </form>
+    </div>
+    <script>
+     // Add data from JS table to qc bulk edit form before submit
+     let qcForm = document.getElementById("qc-form-control");
+     qcForm.addEventListener("submit", (e) => {
+         e.preventDefault();
+         addSelectedSamplesToQcForm(qcForm)
+         qcForm.submit()
+     });
+
+     /**
+      * Adds selected samples to HTML form by creating and appending
+      * hidden input elements with the sample IDs.
+      *
+      * @param   {HTMLFormElement}  form    The HTML form element to which the selected samples will be added.
+      */
+     function addSelectedSamplesToQcForm(form) {
+         // Retrieve selected samples from sessionStorage
+         var selectedSamples = window.getSelectedRows()
```

Comment:
```suggestion
         let selectedSamples = window.getSelectedRows()
```

Avoid using `var` when its not necessary because they can be accidentally re declared. Its often better to use `let` or `const`